### PR TITLE
Remove title bar in macOS.

### PR DESCRIPTION
**Pull request by adamjspooner**
This removes the title bar from macOS, while retaining draggability and the close, minimize, and maximize window options.

screen shot 2018-05-08 at 10 00 17 am

### CODE REVIEW

**Comment by adamjspooner** [comment]
@chrisnojima, what problems did we run into last time we tried to do this?

File: shared/desktop/renderer/index.js [hunk 1]
```diff
@@ -155,7 +155,7 @@ function render(store, MainComponent) {
   }
   ReactDOM.render(
     <Root store={store}>
-      <div style={{display: 'flex', flex: 1}}>
+      <div style={{display: 'flex', flex: 1, '-webkit-app-region': 'drag'}}>
```
Comment (chrisnojima) [inline shared/desktop/renderer/index.js]: we have a style for this: https://github.com/keybase/client/blob/master/shared/styles/index.desktop.js#L76

**Comment by chrisnojima** [comment]
Originally electron had a bug where hover in the title bar area didn't register. so if you had a hover color on the buttons it wouldn't work. We haven't revisited recently so its possible they fixed it.
Also nothing about this pr gates this to just osx so i'm assuming this affect linux and windows incorrectly

**Comment by adamjspooner** [comment]
Let me spin up some virtual machines to test on other platforms.

File: shared/desktop/renderer/index.js (hunk 1 above)
Comment (adamjspooner) [reply shared/desktop/renderer/index.js]: Nice!

**Comment by adamjspooner** [comment]
After much weeping and gnashing of teeth, I was able to get this branch running on Ubuntu and Windows 10. These settings don't affect either, only macOS.

That said, my placement of `-webkit-app-region: 'drag'` will need to be finessed before asking for another review. Tapping and dragging on words in a chat (for highlighting to copy) drags the entire window, as you'd expect with that style.

**Comment by adamjspooner** [comment]
@chrisnojima, this is ready for another review. I've placed window dragging where it makes sense (headers, the tab bar, etc.). Hover styles in the title bar area work, with one exception: `cursor: "pointer"` has no affect. Otherwise, colors change on hover (not sure what else to check). Is that a showstopper?

**Comment by adamjspooner** [comment]
Just noticed the Flow error. I'll fix that tomorrow.

**Comment by adamjspooner** [comment]
@chrisnojima, this is ready for another review.

**Comment by adamjspooner** [comment]
I'm closing this based on Nojima's concern about needing to add the drag styles for every possible view.

**Pull request closed without merging**